resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/Config.scala

```diff
@@ -147,26 +147,30 @@ private[spark] object Config extends Logging {
       .createWithDefault(0)
 
   object ExecutorRollPolicy extends Enumeration {
-    val ID, ADD_TIME, TOTAL_GC_TIME, TOTAL_DURATION, AVERAGE_DURATION, FAILED_TASKS = Value
+    val ID, ADD_TIME, TOTAL_GC_TIME, TOTAL_DURATION, AVERAGE_DURATION, FAILED_TASKS, OUTLIER = Value
   }
 
   val EXECUTOR_ROLL_POLICY =
     ConfigBuilder("spark.kubernetes.executor.rollPolicy")
-      .doc("Executor roll policy: Valid values are ID, ADD_TIME, TOTAL_GC_TIME (default), " +
-        "TOTAL_DURATION, and FAILED_TASKS. " +
+      .doc("Executor roll policy: Valid values are ID, ADD_TIME, TOTAL_GC_TIME, " +
+        "TOTAL_DURATION, FAILED_TASKS, and OUTLIER (default). " +
         "When executor roll happens, Spark uses this policy to choose " +
         "an executor and decommission it. The built-in policies are based on executor summary." +
         "ID policy chooses an executor with the smallest executor ID. " +
         "ADD_TIME policy chooses an executor with the smallest add-time. " +
         "TOTAL_GC_TIME policy chooses an executor with the biggest total task GC time. " +
         "TOTAL_DURATION policy chooses an executor with the biggest total task time. " +
         "AVERAGE_DURATION policy chooses an executor with the biggest average task time. " +
-        "FAILED_TASKS policy chooses an executor with the most number of failed tasks.")
+        "FAILED_TASKS policy chooses an executor with the most number of failed tasks. " +
+        "OUTLIER policy chooses an executor with outstanding statistics which is bigger than" +
+        "at least two standard deviation from the mean in average task time, " +
+        "total task time, total task GC time, and the number of failed tasks if exists. " +
+        "If there is no outlier, it works like TOTAL_DURATION policy.")
```

Comment:
previously the default is TOTAL_GC_TIME, why don't we fallback to it?

Reply:
Yes, we experimented it and found that new policy is better in the production.
The worst case example is that some executors generates long hanging tasks always due to user task misconfigurations. In that case, the tasks are cancelled with speculative executions, but GC_TIME is nearly 0 because the executor's tasks do nothing.